addons/links/src/preview.ts

```diff
@@ -36,51 +33,63 @@ const generateUrl = (id: string) => {
 const valueOrCall = (args: string[]) => (value: string | ((...args: string[]) => string)) =>
   typeof value === 'function' ? value(...args) : value;
 
+// Similar to `useStoryContext` from '@storybook/addons', except works outside a decorator/story
+// NOTE: this means it only tells us the 'global' rendered story, and will not do anything
+// sensible if more than one story is rendered to the screen (i.e. docs)
+function getStoryContext(): StoryContext {
+  // eslint-disable-next-line no-underscore-dangle
+  const { previousStory } = global.__STORYBOOK_PREVIEW__ as PreviewWeb;
+  return global.__STORYBOOK_STORY_STORE__.getStoryContext(previousStory);
+}
+
+function getStoryIndex(): StoryIndex {
+  return global.__STORYBOOK_STORY_STORE__.storyIndex;
+}
+
 export const linkTo = (
-  idOrKindInput: string,
-  storyInput?: string | ((...args: any[]) => string)
+  idOrTitleInput: string,
+  nameInput?: string | ((...args: any[]) => string)
 ) => (...args: any[]) => {
   const resolver = valueOrCall(args);
-  const { storyId } = storyStore.getSelection();
-  const current = storyStore.fromId(storyId) || {};
-  const kindVal = resolver(idOrKindInput);
-  const storyVal = resolver(storyInput);
+  const current = getStoryContext();
+  const titleVal = resolver(idOrTitleInput);
+  const nameVal = resolver(nameInput);
 
-  const fromid = storyStore.fromId(kindVal);
+  const { stories } = getStoryIndex();
+  const fromid = stories[titleVal];
```

Comment:
This line doesn't make sense to me.
- `fromid` => `fromId`
- `titleVal` => `idOrTitleVal` ?